### PR TITLE
Add grant types

### PR DESCRIPTION
**Pull request by faissalMT**
**What**  
Adds grant types to the admin page
Screenie_13

**Why**  
There are multiple separate grants that can be awarded

Also includes some refactoring mostly in the hopes of making future unit testing easier, but also to improve the general malleability of the codebase.

### CODE REVIEW

File: components/ApplicationGrantAmountSelector/ApplicationGrantAmountSelector.test.jsx [hunk 1]
```diff
@@ -0,0 +1,121 @@
+import {
+  /*ApplicationGrantAmountSelector,*/ handleOnChange,
```
Comment (yusufsheiqh) [inline components/ApplicationGrantAmountSelector/ApplicationGrantAmountSelector.test.jsx:2]: Let's remove this if we are not using it.

```suggestion
  handleOnChange,
```

File: components/ApplicationGrantAmountSelector/ApplicationGrantAmountSelector.test.jsx [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+        onChangeSpy,
+        grantApplicationPatcherSpy,
+        applicationId,
+        storeAs
+      )(grantAmountAwarded);
+
+      expect(setErrorSpy).toHaveBeenCalledWith(validationErrorMessage);
+    });
+
+    it('Unsets the error if successful', async () => {
```
Comment (yusufsheiqh) [inline components/ApplicationGrantAmountSelector/ApplicationGrantAmountSelector.test.jsx:78]: Doesn't it actually get set to false before the `try...catch` statement?

File: db/migrations/sqls/20201201100654-add-new-grants-up.sql [hunk 3]
```diff
@@ -0,0 +1,4 @@
+alter table application_assessment drop column grant_amount_awarded;
+alter table application_assessment add column lrsg_closed_businesses_amount decimal(12,2) default 0;
+alter table application_assessment add column lrsg_sector_amount decimal(12,2) default 0;
+alter table application_assessment add column lrsg_open_amount decimal(12,2) default 0;
```
Comment (yusufsheiqh) [inline db/migrations/sqls/20201201100654-add-new-grants-up.sql:4]: Why do we have a decimal precision of 12 here?

**Review by yusufsheiqh — COMMENTED** [review]
Only a few minor questions, but otherwise it looks good to me. I like the addition of tests!

File: db/migrations/sqls/20201201100654-add-new-grants-up.sql (hunk 3 above)
Comment (faissalMT) [reply db/migrations/sqls/20201201100654-add-new-grants-up.sql:4]: Just so we have wiggle-room to use larger numbers if necessary

File: components/ApplicationGrantAmountSelector/ApplicationGrantAmountSelector.test.jsx (hunk 2 above)
Comment (faissalMT) [reply components/ApplicationGrantAmountSelector/ApplicationGrantAmountSelector.test.jsx:78]: Yes, but that value will be overwritten if an exception is caught

File: db/migrations/sqls/20201201100654-add-new-grants-up.sql (hunk 3 above)
Comment (yusufsheiqh) [reply db/migrations/sqls/20201201100654-add-new-grants-up.sql:4]: That is fair, although I do feel 12 is large precision, but it doesn't matter too much here.

File: components/ApplicationGrantAmountSelector/ApplicationGrantAmountSelector.test.jsx (hunk 2 above)
Comment (yusufsheiqh) [reply components/ApplicationGrantAmountSelector/ApplicationGrantAmountSelector.test.jsx:78]: The error is only set if an exception is caught, no? I thought the default value is set to false, so the error is not unset. That is my understanding of it, at least.

**Review by yusufsheiqh — APPROVED** [review]

**Review by mattbee — COMMENTED** [review]
Looks good, just 2 minor comments (but see it's merged so all good for me!)

**Pull request merged**